From 24feca1579b6bdcff4d9b495ce5ce0da6b4b6dc0 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Tue, 28 Feb 2017 16:10:39 +0100
Subject: [PATCH] ip: return correct exit code on route failure

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1417289
Upstream Status: iproute2.git commit d58ba4ba2a532
Conflicts: Previously backported changes to rtnl_talk() signature.

commit d58ba4ba2a532f3ebfa640c42ec2622b54740cdf
Author: Stephen Hemminger <shemming@brocade.com>
Date:   Wed May 6 09:48:06 2015 -0700

    ip: return correct exit code on route failure

    If kernel complains about ip route request, exit status should be
    2 not 1.

    This fixes regression introduced by:
    commit 42ecedd4bae534fc688194a795eb4548c6530cda
    Author: Roopa Prabhu <roopa@cumulusnetworks.com>
    Date:   Tue Mar 17 19:26:32 2015 -0700

        fix ip -force -batch to continue on errors
---
 ip/iproute.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ip/iproute.c b/ip/iproute.c
index 3f4b91a..73c6db9 100644
--- a/ip/iproute.c
+++ b/ip/iproute.c
@@ -1114,7 +1114,7 @@ static int iproute_modify(int cmd, unsigned flags, int argc, char **argv)
 		req.r.rtm_family = AF_INET;
 
 	if (rtnl_talk(&rth, &req.n, NULL, 0) < 0)
-		return -1;
+		return -2;
 
 	return 0;
 }
-- 
1.8.3.1

